src/Templates/BlockTemplatesCompatibility.php

```diff
@@ -0,0 +1,277 @@
+<?php
+namespace Automattic\WooCommerce\Blocks\Templates;
+
+/**
+ * BlockTemplatesCompatibility class.
+ *
+ * To bridge the gap on compatibility with PHP hooks and blockified templates.
+ *
+ * @internal
+ */
+class BlockTemplatesCompatibility {
+
+	const LOOP_ITEM_ID = 'product-loop-item';
+
+	/**
+	 * Hoding the supported hooks, their positions and the hooked functions.
+	 *
+	 * @var array $hook_data
+	 */
+	protected $hook_data = array(
+		'core/query'            => array(
+			'woocommerce_before_main_content' => array(
+				'position' => 'before',
+				'hooked'   => array(
+					'woocommerce_output_content_wrapper' => 10,
+					'woocommerce_breadcrumb'             => 20,
+				),
+			),
+			'woocommerce_after_main_content'  => array(
+				'position' => 'after',
+				'hooked'   => array(
+					'woocommerce_output_content_wrapper_end' => 10,
+				),
+			),
+		),
+		'core/post-title'       => array(
+			'woocommerce_before_shop_loop_item_title' => array(
+				'position' => 'before',
+				'hooked'   => array(
+					'woocommerce_show_product_loop_sale_flash' => 10,
+					'woocommerce_template_loop_product_thumbnail' => 10,
+				),
+			),
+			'woocommerce_shop_loop_item_title'        => array(
+				'position' => 'after',
```

Comment:
When testing this action I noticed it behaves differently between a classic theme and a block theme if no priority is set:

Storefront | TT3
--- | ---
![imatge](https://user-images.githubusercontent.com/3616980/212626144-f20649d3-32d6-4204-9020-c764b5f056a9.png) | ![imatge](https://user-images.githubusercontent.com/3616980/212626249-83313a14-ca03-447f-9c44-510ab264d87c.png)
_(Hook is rendered before the title)_ | _(Hook is rendered after the title)_

Code used:
```PHP
add_action(
	'woocommerce_shop_loop_item_title',
	function () {
		echo 'Hook: woocommerce_shop_loop_item_title';
	}
);
```

Do you think we should change the `position` to `before` so it works the same with actions with default priority?

I'm also wondering if we should support this hook at all. `woocommerce_before_shop_loop_item_title` and `woocommerce_after_shop_loop_item_title` might probably be enough for all possible usecases in block templates.

Reply:
I changed the position to `before` as you suggested. 

> I'm also wondering if we should support this hook at all. `woocommerce_before_shop_loop_item_title` and `woocommerce_after_shop_loop_item_title` might probably be enough for all possible usecases in block templates.

This action is used 21 times across WC.com extensions. Given that we haven't analyzed other extensions' marketplace, I think we should still support it. What do you think?